substratevm/src/com.oracle.svm.configure.jdk11/src/com/oracle/svm/configure/filters/ModuleFilterToolsJDK11OrLater.java

```diff
@@ -29,11 +29,17 @@
 import java.util.HashSet;
 import java.util.Set;
 
+import org.graalvm.compiler.serviceprovider.JavaVersionUtil;
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.hosted.Feature;
+
 import com.oracle.svm.configure.filters.RuleNode.Inclusion;
+import com.oracle.svm.core.annotate.AutomaticFeature;
 
-public class ModuleFilterTools {
+public class ModuleFilterToolsJDK11OrLater implements ModuleFilterTools {
 
-    public static RuleNode generateFromModules(String[] moduleNames, Inclusion rootInclusion, Inclusion exportedInclusion, Inclusion unexportedInclusion, boolean reduce) {
+    @Override
+    public RuleNode generateFromModules(String[] moduleNames, Inclusion rootInclusion, Inclusion exportedInclusion, Inclusion unexportedInclusion, boolean reduce) {
```

Comment:
As @peter-hofer pointed out this does not work because

1. com.oracle.svm.configure.ConfigurationTool#generateFilterRules does onnlt make sense when the ConfigurationTool runs on the JVM.
2. When ConfigurationTool runs on the JVM `ImageSingletons` and `@AutomaticFeature` is not available.

Reply:
Fixed in https://github.com/oracle/graal/pull/3546/commits/695c01944011b4e9a45ea4f6263afc4c77530a9f